app/models/components/course/duplicator_component_host.rb

```diff
@@ -0,0 +1,15 @@
+class Duplicator::ModelComponentHost
```

Comment:
Your class cannot be autoloaded because it does not follow Rails conventions.

But before that, should you be declaring a new host, or should you piggyback on the course model host?